### PR TITLE
Separate bootstrap building logic into the envoy/bootstrap package

### PR DESCRIPTION
**Pull request by steeling**
This PR separates the bootstrap config build logic into the bootstrap package. There are 0 "logic" code changes, only moving code around, exporting previously unexported functions, and prefacing methods/types with the new package name.

The main reason to do this is that there are changes coming with respect to rotating the secrets with the bootstrap config. Because of this, we will need to modify how we are creating and writing the bootstrap config.

Currently the code is tightly coupled with generating the config and writing it. This PR begins the process of decoupling, and brings us closer the the [builder pattern](https://docs.google.com/document/d/1L_u-XlVU8Fog-uFS3Uf0O2nz8IArQy82cd6Npxm038Y/edit) we plan on implementing.

The main changes include:

1. Move HealthProbes, which are an internal construct, to a separate models package, so that it can be used by both the injector and the bootstrap builder.
2. Move all code that is involved in building the bootstrap config to the bootstrap package. A subsequent PR will make actual changes to create a logical builder.

### CODE REVIEW

**Comment by steeling** [comment]
> Can this be noted as part of #4839?

This would actually be part of #4818

**Review by nojnhuh — APPROVED** [review]

File: pkg/injector/bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy/bootstrap"
+	"github.com/openservicemesh/osm/pkg/errcode"
+	"github.com/openservicemesh/osm/pkg/models"
+	"github.com/openservicemesh/osm/pkg/utils"
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+// This will read an existing envoy bootstrap config, and create a new copy by changing the NodeID, and certificates.
```
Comment (shalier) [inline pkg/injector/bootstrap.go:20]:
```suggestion
// createEnvoyBootstrapFromExisting will read an existing envoy bootstrap config, and create a new copy by changing the NodeID, and certificates.
```

File: pkg/injector/bootstrap.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		TLSMinProtocolVersion: wh.configurator.GetMeshConfig().Spec.Sidecar.TLSMinProtocolVersion,
+		TLSMaxProtocolVersion: wh.configurator.GetMeshConfig().Spec.Sidecar.TLSMaxProtocolVersion,
+		CipherSuites:          wh.configurator.GetMeshConfig().Spec.Sidecar.CipherSuites,
+		ECDHCurves:            wh.configurator.GetMeshConfig().Spec.Sidecar.ECDHCurves,
+	}
+	config, err := bootstrap.GenerateEnvoyConfig(configMeta, wh.configurator)
+	if err != nil {
+		return nil, err
+	}
+	// marshalAndSaveBootstrap
```
Comment (shalier) [inline pkg/injector/bootstrap.go:60]: Nit
```suggestion

```

File: pkg/models/healthprobe.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	// isHTTP corresponds to an httpGet probe with a scheme of HTTP or undefined.
+	// This helps inform what kind of Envoy config to add to the pod.
+	IsHTTP bool
+
+	// isTCPSocket indicates if the probe defines a TCPSocketAction.
+	IsTCPSocket bool
+}
+
+// HealthProbes is to serve as an indication whether the given healthProbe has been rewritten
```
Comment (shalier) [inline pkg/models/healthprobe.go:19]:
```suggestion
// HealthProbes is to serve as an indication of whether the given healthProbe has been rewritten
```

**Review by shalier — APPROVED** [review]
LGTM

File: pkg/models/healthprobe.go (hunk 3 above)
Comment (steeling) [reply pkg/models/healthprobe.go:19]: done!

**Review by jaellio — APPROVED** [review]

**Pull request merged**